dts/bindings/pinctrl/nxp,kinetis-pinmux.yaml

```diff
@@ -11,6 +11,27 @@ properties:
     clocks:
       required: true
 
-pinmux-cells:
-  - pin
-  - function
+child-binding:
+    description: |
+      NXP Kinetis Pin data.  A series of child nodes that describe each pin
+      configuration supported by the SoC.  Each node is expected to named as
+      follows - <periph>_<signal>_pt<port><pin>
+
+      The node will have a matching node label that board dtsi files can
+      utilized to reference in pinctrl-<N> properties.
+
+      The following is an example for UART0 CTS signal on pin 0, with mux
+      value of 2.
+
+      uart0_cts_pta0: uart0_cts_pta0 {
+           nxp,kinetis-port-pins = < 0 2 >;
```

Comment:
I'm curious how you plan to extract the port information so the pin/func is conveyed to the right gpio device.

Reply:
Using `DT_PARENT()`

Something like for getting the port struct device:

```
#define GET_PORT_DEV(i, node) \
        DEVICE_DT_GET(DT_PARENT(DT_PHANDLE_BY_IDX(node, pinctrl_0, i)))
```

Reply:
Ok, so the nodes would be child nodes of gpio devices and not of peripheral devices (uart, spi, ..) ?

Reply:
@erwango yes, the "pin data" will be child nodes of the port device (which is separate from the GPIO device) on the NXP Kinetis SoCs.